### PR TITLE
RSDK-6120: Refactor ModuleService_ -> ModuleService boilerplate

### PR DESCRIPTION
**Pull request by hexbabe**
https://viam.atlassian.net/browse/RSDK-6120

Made changes following C++ SDK examples

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -1,14 +1,13 @@
...
-int main(int argc, char* argv[]) {
+int main(int argc, char** argv) {
     const std::string usage = "usage: camera_realsense /path/to/unix/socket";
 
     if (argc < 2) {
         std::cout << "ERROR: insufficient arguments\n";
         std::cout << usage << "\n";
         return EXIT_FAILURE;
     }
-    std::cout << "About to serve on socket " << argv[1] << std::endl;
```
Comment (hexbabe) [inline main.cpp:11]: Log is now extraneous as C++ SDK logs basically the same thing right after

```1/23/24, 10:21:37.981 AM   info robot_server.process.viam_realsense_/viam-camera-realsense-latest-aarch64.AppImage.StdOut   pexec/managed_process.go:244   \_ [2024-01-23 10:21:37.981218] [0x0000ffffb6a67040] [info] Module listening on /tmp/viam-module-4223363164/viam_realsense.sock  
1/23/24, 10:21:37.322 AM   info robot_server.process.viam_realsense_/viam-camera-realsense-latest-aarch64.AppImage.StdOut   pexec/managed_process.go:244   \_ About to serve on socket /tmp/viam-module-4223363164/viam_realsense.sock  
```

**Comment by hexbabe** [comment]
Hmm. Tests are failing because the CI runner is using the old container using the older version of the C++ SDK. @bhaney 

I remember you said the image/container will update post-merge? But we probably shouldn't ignore the check..

File: src/camera_realsense.cpp [hunk 2]
```diff
@@ -921,46 +921,19 @@ std::vector<std::string> validate(sdk::ResourceConfig cfg) {
...
-
-    auto module_registration = std::make_shared<sdk::ModelRegistration>(
-        sdk::ResourceType{kResourceType}, sdk::Camera::static_api(),
+int serve(int argc, char** argv) {
+    std::shared_ptr<sdk::ModelRegistration> mr = std::make_shared<sdk::ModelRegistration>(
+        sdk::API::get<sdk::Camera>(),
         sdk::Model{kAPINamespace, kAPIType, kAPISubtype},
         [](sdk::Dependencies deps, sdk::ResourceConfig cfg) -> std::shared_ptr<sdk::Resource> {
-            return std::make_shared<CameraRealSense>(deps, cfg);
+            return std::make_unique<CameraRealSense>(deps, cfg);
```
Comment (bhaney) [inline src/camera_realsense.cpp:929]: why the change from make_shared to make_unique?

**Review by bhaney — CHANGES_REQUESTED** [review]
Could you test out these changes by running the integration tests as well on an actual intel realsense and making sure that things work as expected?

File: src/camera_realsense.cpp (hunk 2 above)
Comment (hexbabe) [reply src/camera_realsense.cpp:929]: I think this is me taking the boilerplate format of the complex example in the C++ SDK. I assume it's because there should only be one instance of the CameraRealSense obj? If this is not true for the module we should certainly change it back

**Comment by hexbabe** [comment]
lol I tried for a while but my Ubuntu version is too old for the integration test to build I think

I already tried it out this morning on my D435 + Orin Nano and it works!

Uploading yuh.mov…

**Comment by hexbabe** [comment]
https://github.com/viamrobotics/viam-camera-realsense/assets/55464069/dbf2cf10-4394-4d9d-a0b3-770c10e412a0

**Review by bhaney — APPROVED** [review]
LGTM! I checked to make sure the unique vs. shared pointer wouldn't affect anything, and I don't think it will
